posydon/interpolation/profile_interpolation.py

```diff
@@ -0,0 +1,572 @@
+                testing_failed.append(i)
+                pass
+        warnings.warn(f"{len(testing_failed)} binaries failed")
+
+        # extract training data
+        print("extracting training data")
+        train = PSyGrid(train_path)  # load PSyGrid object for training grid
+        self.scalars = pd.DataFrame()
+        self.profiles = []
+        training_failed = []
+        
+        for i in range(len(train)):
+            try:
+                scalars,profiles = self.scrape(train,i)
+                self.scalars = self.scalars.append(scalars,ignore_index=True)
+                self.profiles.append(profiles)
+            except:
+                training_failed.append(i)
+                pass
+        warnings.warn(f"{len(training_failed)} training binaries failed")
+        
+
+    def scrape(self,grid,ind):
+        """Extracts profile data from one MESA run.
+        Args:
+            grid (obj) : PSyGrid object.
+            ind (int) : index of run to be scraped.
+        Returns:
+            scalars (array-like) : dictionary containing initial 
+                                   m1, m2, p and final s1_state, final_m1.
+            profiles (array-like) : all N specified profiles, shape (N,200).
+        """
+        # open individual run as a DataFrame
+        df = pd.DataFrame(grid[ind]['final_profile1'])
+        df = df.sort_values(by="mass")
+        df = df.reset_index(drop=True)
+
+        # grab input values, final star 1 state, final star 1 mass
+        final_m1 = grid.final_values["star_1_mass"][ind]
+        scalars = {"m1":grid.initial_values["star_1_mass"][ind],
+                   "m2":grid.initial_values["star_2_mass"][ind],
+                   "p":grid.initial_values["period_days"][ind],
+                   "s1_state":grid.final_values["S1_state"][ind],
+                   "final_m1":final_m1}
+
+        # grab output vectors, interpolate to normalize
+        profiles=np.zeros([len(self.names),200])
+        for i,prof in enumerate(self.names):
+            if prof in df.columns:
+            
+                f = interp1d(df['mass']/final_m1,df[prof],
+                             fill_value="extrapolate")
+                profile_new = f(np.linspace(0,1,200))
+                profiles[i] = profile_new
+            else:
+                warnings.warn(f"{prof} profile not saved in grid, will not be included in file")
+
+        return scalars, profiles
+
+    def save(self, filename):
+        """Save extracted profile data.
+        Args:
+            filename (str) : path/name of '.pkl' file where the data will be saved.
+        """
+        SAVE_ATTRS = self.__dict__
+        DONT_SAVE = []
+        myattrs = {key: SAVE_ATTRS[key]
+                   for key in SAVE_ATTRS if key not in DONT_SAVE}
+        with open(filename, 'wb') as f:
+            pickle.dump(myattrs, f)
+            
+            
+class ProfileInterpolator:
+    
+    def __init__(self,load_interpolator=None):
```

Comment:
name this argument filename to keep it consistent with the IFInterpolator and have a separate load method